server/functions/0_catalog.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package functions
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression/function"
+	"github.com/dolthub/go-mysql-server/sql/expression/function/aggregation"
+)
+
+// Function is a name, along with a collection of functions, that represent a single PostgreSQL function with all of its
+// overloads.
+type Function struct {
+	Name      string
+	Overloads []interface{}
+}
+
+// Catalog contains all of the PostgreSQL functions. If a new function is added, make sure to add it to the catalog here.
+var Catalog = []Function{
+	cbrt,
+	gcd,
+	lcm,
+	round,
+}
+
+// init handles the initialization of the catalog by overwriting the built-in GMS functions, since they do not apply to
+// PostgreSQL (and functions of the same name often have different behavior).
+func init() {
```

Comment:
This is overly aggressive. While we're building out function support, leave the mysql ones alone unless there's a name conflict. It's more useful for the purpose of increasing test coverage / verifying correctness to have the MySQL functions available (esp things like aggregates, which work mostly the same)